src/struct.c

```diff
@@ -631,6 +631,52 @@ Dsymbol *AggregateDeclaration::searchCtor()
     return s;
 }
 
+/***************************************
+ * Return true if aggregate is an InputRange.
+ */
+bool AggregateDeclaration::isInputRange()
+{
+    return (this->search(Loc(), Id::Ffront) &&
```

Comment:
Doesn't this simply check if the aggregate has a member `front`? Simply having members named `front`, `popFront` and `empty` does not make an input range: http://dlang.org/phobos/std_range#isInputRange